src/Relation/Binary/PropositionalEquality/Core.agda

```diff
@@ -66,6 +66,18 @@ respʳ _∼_ refl x∼y = x∼y
 resp₂ : ∀ (∼ : Rel A ℓ) → ∼ Respects₂ _≡_
 resp₂ _∼_ = respʳ _∼_ , respˡ _∼_
 
+------------------------------------------------------------------------
+-- Some implicit variants of refl and cong; 'i' for implicit
+irefl : {f : A → B} (x : A) → f x ≡ f x
+irefl _ = refl
+
+icong : ∀ {f : A → B} {x y} → x ≡ y → f x ≡ f y
+icong = cong _
+
+-- ′ for simplified à la ∘′
+icong′ : ∀ {f : A → B} x → f x ≡ f x
+icong′ {f = f} x = cong f refl
```

Comment:
I have to confess I'm struggling to see the difference between `irefl` and `icong′`...

Reply:
Indeed. I didn't look hard enough when I grabbed stuff from that thread! I definitely prefer the name `irefl` though.

Reply:
I think `irefl` should drop the `f` argument and be `erefl`
(for 'explicit' `refl`) and be defined via a pattern synonym
so it is available both on the left & on the right.

Reply:
I don't quite understand: `irefl` is a function, so defining `erefl` as a pattern would be trying to match against a function when used on the left, no?

Reply:
Also, you can't drop the `f` argument, as you want to provide an explicit input at one type, and then have the equality be at another. That's how it shows up in lots of applications.